server/v2/config.go

```diff
@@ -23,6 +22,12 @@ func ReadConfig(configPath string) (*viper.Viper, error) {
 		return nil, fmt.Errorf("failed to merge configuration: %w", err)
 	}
 
+	// Read client.toml to get chain-id
+	v.SetConfigName("client")
```

Comment:
mmh, I don't think server should read client config.
Especially that it isn't set by default.

In server v0 we read it from genesis or extract it from state. Cannot we do the same in the comet server directly?